OrbitGl/TimerTrack.h

```diff
@@ -27,7 +27,7 @@ class TimerTrack : public Track {
 
   // Pickable
   void Draw(GlCanvas* canvas, PickingMode picking_mode) override;
-  void OnTimer(const orbit_client_protos::TimerInfo& timer_info);
+  virtual void OnTimer(const orbit_client_protos::TimerInfo& timer_info);
```

Comment:
Just want to say that this and `AsyncTrack` show how this is hardly is-a inheritance, but rather a way to share code with the parent class while overwriting some methods. Much bigger than this review, probably never-to-be-addressed, and maybe normal for this kind of ui work, but the relationship between these classes is getting more and more confusing.

Reply:
An AsyncTrack deals with timers (in the sense of time spans), why is it not a TimerTrack conceptually? Is the "TimerTrack" name the problem? The way I see it, AsyncTrack very much is-a TimerTrack, or a SpanTrack maybe!?

Reply:
Agree with @pierricgimmig here, but I also agree one has to be careful with (virtual) inheritance. It's easy to create a class hierarchy that is hard to use and where the primary reason for its existence is sharing of code. 

But I think there are some good reasons for having it the way it is done right now:
- The base class Track allows TimeGraph to just draw a track without case distinction (with some odd, minor exceptions). 
-  For users (person writing a new track), having TimerTrack actually makes things very easy and straightforward to do. In most cases you can start using TimerTrack and then gradually customize what you need (like tooltips). More generally, TimerTrack provides a framework how to write a new timer-based track.
- And, yes, reuse of code (UpdatePrimitives) is a reason here. I'm very much in favor of a functional style where UpdatePrimitives would just be a free function, but it would require to add a good amount of customization options (maybe passing functions into UpdatePrimitives?).

@dpallotti If you have suggestions how this can be improved, I'd be happy to discuss.

Reply:
> An AsyncTrack deals with timers (in the sense of time spans), why is it not a TimerTrack conceptually?

Inheritance is not about the concept, e.g., the famous "a square is not a rectangle". Rather it's about about the behaviour, and this usually fails when using overriding to replace some behavior of the parent in order to share the code of other methods. That was my point. That's why in most cases a methods should either be pure virtual or final.

But this was just a general thought and not strictly related to this pull request. I realize how ui classes are very tricky in this regard and with so many methods adhering to a concept just for the sake of it makes things unnecessarily complicated. But it's something to keep in mind. E.g., sometimes instead of having a subclass replace a method of the superclass, you should have a superclass with two subclasses that express the two behaviors.